onnx/helper.py

```diff
@@ -277,6 +277,16 @@ def split_complex_to_pairs(ca: Sequence[np.complex64]) -> Sequence[int]:
             for i in range(len(ca) * 2)]
 
 
+# convert a f32 (as uint32) to bf16 (as uint32)
+# TODO: big-endian
```

Comment:
FWIW: there is some code that does a cast here:
https://github.com/onnx/onnx/blob/5b1346e30d66e4ec550f6b63c3883b258a2e8e3e/onnx/backend/test/case/node/cast.py#L37
May be worth unifying into some common code (especially if there is some minor difference).

Reply:
that code is wrong; it's not rounding properly when converting from F32 to BF16. Particularly this:
` np_bfp16 = np_uint16_view[1::2] if little_endisan else np_uint16_view[0::2]`

It looks like this is all tests code though, so its fine for testing as long as the test values are crafted in such a way as to not need rounding.

Reply:
yes, I saw the difference. That's why I was suggesting unifying both to use a common *correct* implementation.